metals/src/main/scala/scala/meta/internal/metals/watcher/SimpleFileWatcher.scala

```diff
@@ -0,0 +1,66 @@
+package scala.meta.internal.metals.watcher
+
+import collection.JavaConverters._
+import java.nio.file.{FileSystems, Path, StandardWatchEventKinds}
+import java.util.concurrent.BlockingQueue
+import scala.concurrent.{ExecutionContext, Future}
+
+object SimpleFileWatcher {
```

Comment:
I know testing file watching end-to-end is probably difficult, so you may want to look into how testing is done in this repository to see if we can test this component in isolation.